.gitignore

```diff
@@ -2,4 +2,8 @@
 appengine-generated/
 
 # maven
-target/
\ No newline at end of file
+target/
+
+# frontend
+.cache/
+node_modules/
```

Comment:
I would have love not to upload package-lock but its necessary:
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5